### ISSUE TITLE
Git Bash: Critical Error(VM_COMMAND: end)

### ISSUE THREAD

**Issue by GatesGe** [git-for-windows/git#4065]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options

git version 2.38.0.windows.1
cpu: x86_64
built from commit: 0355083fbe5582f6d3f819afc130ed2f2375e0bd
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Windows 10 Pro for Workstations X64 [Version 10.0.19044.2130]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"

# C:\Program Files\Git\etc\install-options.txt
Editor Option: VIM
Custom Editor Path: 
Default Branch Option: main
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Merge
Use Credential Manager: Enabled
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
Enable FSMonitor: Disabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

    **After the Git Bash is opened in `options -> terminal`, it always reports a Critical Error ( VM_COMMAND: end ) , and no matter whether you click close or OK, it will cause a forced exit. This phenomenon persisted for several versions, and I don't know what caused it.**

![Critical Error](https://camo.githubusercontent.com/a08f9498c87a77da8f9f4429aa8ce525890ba2e684fcbfcaea49ee86e63d4b94/68747470733a2f2f70312e6d65697475616e2e6e65742f6470706c6174666f726d2f616135356366366361643532383939333366323031323364616435636332613934343239372e706e67)

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

    **Git Bash**

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
    # command not executed
```
 - What did you expect to occur after running these commands?

    **null**

 - What actually happened instead?

    **null**

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

    **null**

**Comment by rimrul** [comment]
Looks like a [segfault in MinTTY](https://github.com/mintty/mintty/blob/389dc7791f45c1329011bd47b658f68ffcd2d4c8/src/windialog.c#L185-L192) after a recent [`WM_COMMAND` or `WM_DRAWITEM`](https://github.com/mintty/mintty/blob/389dc7791f45c1329011bd47b658f68ffcd2d4c8/src/windialog.c#L698-L706) was processed by the config dialog is what causes that messagebox. (The weird `when`/`or` syntax is basically just `case`](https://github.com/mintty/mintty/blob/389dc7791f45c1329011bd47b658f68ffcd2d4c8/src/std.h#L145-L147))

**Comment by GatesGe** [comment]
> Looks like a [segfault in MinTTY](https://github.com/mintty/mintty/blob/389dc7791f45c1329011bd47b658f68ffcd2d4c8/src/windialog.c#L185-L192) after a recent [`WM_COMMAND` or `WM_DRAWITEM`](https://github.com/mintty/mintty/blob/389dc7791f45c1329011bd47b658f68ffcd2d4c8/src/windialog.c#L698-L706) was processed by the config dialog is what causes that messagebox. ([The weird `when`/`or` syntax is basically just `case`](https://github.com/mintty/mintty/blob/389dc7791f45c1329011bd47b658f68ffcd2d4c8/src/std.h#L145-L147))

I found that the cause of this error is not directly caused by the source code, but caused by Windows updating drivers or disabling system services. After reinstalling windows, I installed git-bash without any settings, and no error was reported when I opened this setting; but when I updated the driver and optimized system services, and then installed git-bash, this error appeared. This factor was not discovered at first, and was discovered later, but I don't know what went wrong.

**Comment by rimrul** [comment]
An issue with the same symptoms was reported to mintty as mintty/mintty#1185 but they did not manage to debug the issue further.

**Comment by GatesGe** [comment]
During use, it is found that this error has nothing to do with the version, it is the Windows system service [Print Spooler]: (C:\Windows\System32\spoolsv.exe) is not started or is disabled or set to manual mode. When this setting is turned on, when the service is called, it will not wake up automatically, causing bugs. The guess may be that the developer did not do more processing on this service exception or did not consider the service being disabled, and regarded it as a key program, and this bug may only be triggered in rare cases.

**Comment by dscho** [comment]
@GatesGe thank you so much for leaving this helpful record in this ticket. I am sure that others encountering the same issue will be very, very grateful.

**Comment by mintty** [comment]
Thanks indeed. Stopping the Print Spooler service makes the bug appear and I can now debug it.

**Comment by dscho** [comment]
This ticket is addressed via https://github.com/git-for-windows/git/issues/4182. @GatesGe would you mind testing once we get a new [snapshot](https://wingit.blob.core.windows.net/files/index.html)?

**Comment by dscho** [comment]
> This ticket is addressed via #4182. @GatesGe would you mind testing once we get a new [snapshot](https://wingit.blob.core.windows.net/files/index.html)?

There's [a new snapshot now](https://wingit.blob.core.windows.net/files/index.html). @GatesGe please verify that it fixes the issue. I plan on releasing a new Git for Windows version soon, and it would be a shame if the bug you reported wasn't fixed in it solely due to lack of testing.

**Comment by dscho** [comment]
/add release note bug A bug in MinTTY caused it to throw a Critical Error when the printer spool service was not started, which was fixed by [upgrading MinTTY](https://github.com/git-for-windows/git/issues/4182).
